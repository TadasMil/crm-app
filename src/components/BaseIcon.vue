<script setup lang="ts">
import { computed } from 'vue'

const props = defineProps({
  icon: {
    type: String,
    default: '',
  },
})

const iconPath = computed(() => `/icons/dashboard.svg?#${props.icon}`)
</script>

<template>
  <svg class="block flex-shrink-0 fill-current">
    <use :href="iconPath" :xlink:href="iconPath" />
  </svg>
</template>
