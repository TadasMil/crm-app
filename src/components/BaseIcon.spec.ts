import { render } from '@testing-library/vue'
import { describe, expect, it } from 'vitest'

import BaseIcon from './BaseIcon.vue'

describe('BaseIcon.spec.js', () => {
  it('should render icon', () => {
    const { html } = render(BaseIcon, {
      props: {
        icon: 'user',
      },
    })

    expect(html()).toMatchSnapshot()
  })
})
