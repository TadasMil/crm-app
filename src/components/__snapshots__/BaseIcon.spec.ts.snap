// Vitest Snapshot v1

exports[`BaseIcon.spec.js > should render icon 1`] = `
"<svg class=\\"block flex-shrink-0 fill-current\\">
  <use href=\\"/icons/dashboard.svg?#user\\" xlink:href=\\"/icons/dashboard.svg?#user\\"></use>
</svg>"
`;
